fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java

```diff
@@ -1150,9 +1206,13 @@ public AlterJobV2 analyzeAndCreateJob(List<AlterClause> alterClauses, Database d
             } else if (alterClause instanceof ReorderColumnsClause) {
                 // reorder column
                 if (olapTable.getKeysType() == KeysType.PRIMARY_KEYS) {
-                    throw new DdlException("Primary key table do not support reorder column");
+                    List<Integer> sortKeyIdxes = new ArrayList<>();
+                    processReorderColumnOfPrimaryKey((ReorderColumnsClause) alterClause, olapTable, indexSchemaMap, sortKeyIdxes);
+                    return createJobForProcessReorderColumnOfPrimaryKey(db.getId(), olapTable,
+                            indexSchemaMap, sortKeyIdxes);
```

Comment:
what if a user made a mistake and use the old usage of `alter table order by`, for example:

```
create table table0 (a,b,c,d,e) primary key(a), order by(c,d) 

alter table order by (d, c, b, c, e)
```
Could we provide a notice of potential misusage if the user provides all the columns in the table in `alter table order by` for primary key table?

Reply:
old usage 1. need all columns. 2. all keys need be the prefix of all columns. 3. column can not be duplicate appear.
if write all columns, we can not tell the differences between make all columns to be sort key on purpose and making a mistake to use the old usage.
maybe the furthest the judgement can do is provide a notice when satisfy 1, 2, since that in the new usage 2 is needless.